Transceiver Mid test failing

### Your environment.
  - Version: 4f362ee

### Steps to reproduce

```
$ go test ./ -run Transceiver_Mid -v -race -count=10
```

### Log

```
=== RUN   TestPeerConnection_Transceiver_Mid
==================
WARNING: DATA RACE
Read at 0x00c00037c410 by goroutine 70:
  github.com/pion/webrtc/v2.(*PeerConnection).startRTPSenders()
      /pion/webrtc/peerconnection.go:973 +0x149
  github.com/pion/webrtc/v2.(*PeerConnection).startRenegotation.func1()
      /pion/webrtc/peerconnection.go:1724 +0x9bc

Previous write at 0x00c00037c410 by goroutine 102:
  github.com/pion/webrtc/v2.(*RTPTransceiver).setSendingTrack()
      /pion/webrtc/rtptransceiver.go:96 +0x7a
  github.com/pion/webrtc/v2.(*PeerConnection).AddTrack()
      /pion/webrtc/peerconnection.go:1196 +0x20c
  github.com/pion/webrtc/v2.TestPeerConnection_Transceiver_Mid()
      /pion/webrtc/peerconnection_renegotation_test.go:297 +0xfa5
  testing.tRunner()
      /usr/local/go/1.14/libexec/src/testing/testing.go:992 +0x1eb

Goroutine 70 (running) created at:
  github.com/pion/webrtc/v2.(*PeerConnection).startRenegotation()
      /pion/webrtc/peerconnection.go:1691 +0x7e
  github.com/pion/webrtc/v2.(*PeerConnection).SetRemoteDescription()
      /pion/webrtc/peerconnection.go:822 +0xb6b
  github.com/pion/webrtc/v2.TestPeerConnection_Transceiver_Mid()
      /pion/webrtc/peerconnection_renegotation_test.go:292 +0xe66
  testing.tRunner()
      /usr/local/go/1.14/libexec/src/testing/testing.go:992 +0x1eb

Goroutine 102 (running) created at:
  testing.(*T).Run()
      /usr/local/go/1.14/libexec/src/testing/testing.go:1043 +0x660
  testing.runTests.func1()
      /usr/local/go/1.14/libexec/src/testing/testing.go:1285 +0xa6
  testing.tRunner()
      /usr/local/go/1.14/libexec/src/testing/testing.go:992 +0x1eb
  testing.runTests()
      /usr/local/go/1.14/libexec/src/testing/testing.go:1283 +0x527
  testing.(*M).Run()
      /usr/local/go/1.14/libexec/src/testing/testing.go:1200 +0x2ff
  main.main()
      _testmain.go:338 +0x223
==================
    TestPeerConnection_Transceiver_Mid: testing.go:906: race detected during execution of test
--- FAIL: TestPeerConnection_Transceiver_Mid (0.43s)
    : testing.go:906: race detected during execution of test
```

--- a/peerconnection_media_test.go
+++ b/peerconnection_media_test.go
@@ -1115,8 +1115,7 @@ func TestPeerConnection_Start_Only_Negotiated_Senders(t *testing.T) {
 	assert.NoError(t, pcOffer.SetRemoteDescription(answer))
 
 	// Wait for senders to be started by startTransports spawned goroutine
-	pcOffer.negotiationLock.Lock()
-	defer pcOffer.negotiationLock.Unlock()
+	waitForNegotiation(pcOffer)
 
 	// sender1 should be started but sender2 should not be started
 	assert.True(t, sender1.hasSent(), "sender1 is not started but should be started")
@@ -1159,10 +1158,7 @@ func TestPeerConnection_Start_Right_Receiver(t *testing.T) {
 
 	assert.NoError(t, signalPair(pcOffer, pcAnswer))
 
-	pcOffer.negotiationLock.Lock()
-	pcAnswer.negotiationLock.Lock()
-	pcOffer.negotiationLock.Unlock()
-	pcAnswer.negotiationLock.Unlock()
+	waitForNegotiation(pcOffer, pcAnswer)
 
 	// transceiver with mid 0 should be started
 	started, err := isTransceiverReceiverStarted(pcAnswer, "0")
@@ -1174,10 +1170,7 @@ func TestPeerConnection_Start_Right_Receiver(t *testing.T) {
 
 	assert.NoError(t, signalPair(pcOffer, pcAnswer))
 
-	pcOffer.negotiationLock.Lock()
-	pcAnswer.negotiationLock.Lock()
-	pcOffer.negotiationLock.Unlock()
-	pcAnswer.negotiationLock.Unlock()
+	waitForNegotiation(pcOffer, pcAnswer)
 
 	// transceiver with mid 0 should not be started
 	started, err = isTransceiverReceiverStarted(pcAnswer, "0")
@@ -1193,10 +1186,7 @@ func TestPeerConnection_Start_Right_Receiver(t *testing.T) {
 
 	assert.NoError(t, signalPair(pcOffer, pcAnswer))
 
-	pcOffer.negotiationLock.Lock()
-	pcAnswer.negotiationLock.Lock()
-	pcOffer.negotiationLock.Unlock()
-	pcAnswer.negotiationLock.Unlock()
+	waitForNegotiation(pcOffer, pcAnswer)
 
 	// transceiver with mid 0 should not be started
 	started, err = isTransceiverReceiverStarted(pcAnswer, "0")

--- a/peerconnection_renegotation_test.go
+++ b/peerconnection_renegotation_test.go
@@ -59,6 +59,13 @@ func sdpMidHasSsrc(offer SessionDescription, mid string, ssrc uint32) bool {
 	return false
 }
 
+func waitForNegotiation(pcs ...*PeerConnection) {
+	for _, pc := range pcs {
+		pc.negotiationLock.Lock()
+		pc.negotiationLock.Unlock() //nolint
+	}
+}
+
 /*
 *  Assert the following behaviors
 * - We are able to call AddTrack after signaling
@@ -268,6 +275,8 @@ func TestPeerConnection_Transceiver_Mid(t *testing.T) {
 	// apply answer so we'll test generateMatchedSDP
 	assert.NoError(t, pcOffer.SetRemoteDescription(answer))
 
+	waitForNegotiation(pcOffer, pcAnswer)
+
 	// Must have 3 media descriptions (2 video and 1 datachannel)
 	assert.Equal(t, len(offer.parsed.MediaDescriptions), 3)
 
@@ -290,6 +299,8 @@ func TestPeerConnection_Transceiver_Mid(t *testing.T) {
 	assert.NoError(t, pcAnswer.SetLocalDescription(answer))
 	// apply answer so we'll test generateMatchedSDP
 	assert.NoError(t, pcOffer.SetRemoteDescription(answer))
+
+	waitForNegotiation(pcOffer, pcAnswer)
 
 	track3, err := pcOffer.NewTrack(DefaultPayloadTypeVP8, rand.Uint32(), "video", "pion3")
 	require.NoError(t, err)

